### PR TITLE
✨ Subtitles

### PR DESCRIPTION
**Pull request by pajowu**
This PR adds support for exporting the transcript as subtitles. They can either be exported as a WebVTT (with speaker names)&SRT or added to the video exports (either by burning them into the video or by adding them as subtitles if the output format is supported (mp4 or mkv):
Screenshot 2022-01-07 at 04 01 27
Screenshot 2022-01-07 at 04 01 45

This PR also includes a few bugfixes:
1. In the export page the output path was sometimes miscalculated. This adds a better calculation and good defaults
2. In #202 a bug was introduced, which lead to the app not starting in production

### CODE REVIEW

File: app/src/core/webvtt.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  limitLineLength: number | null,
+  format: SubtitleFormat
+): Promise<void> {
+  const vttString = contentToVtt(
+    content,
+    wordTimings,
+    includeSpeakerNames,
+    limitLineLength
+  ).toString(format);
+  console.log(vttString);
```
Comment (anuejn) [inline app/src/core/webvtt.ts:93]: do we really want to log this here?

File: app/src/pages/Editor/ExportDocumentDialog.tsx [hunk 2]
```diff
@@ -1,109 +1,105 @@
...
-interface finalcutExport {
-  type: 'OpenTimelineIO';
-  path: string;
-}
-type ExportType = audioExport | videoExport | finalcutExport;
+type ExportType = {
+  type: string;
+  defaultExtension: string;
+  filters: FileFilter[];
+  // We pass a ref as onExport which the function will set to its export function
```
Comment (anuejn) [inline app/src/pages/Editor/ExportDocumentDialog.tsx:22]: "which the type specific export component ..."

File: app/src/util/WebVtt.spec.ts [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+//   vertical?: 'rl' | 'lr';
+//   line?: number | string;
+//   size?: string;
+//   align?: 'start' | 'center' | 'end';
+//
+//   toString() {
+//     return Object.entries(this).map((k, v) => `${k}:${v}`).join(' ')
+//   }
+// }
+import { VttComment, VttCue, VttCueSettings, WebVtt } from './WebVtt';
```
Comment (anuejn) [inline app/src/util/WebVtt.spec.ts:11]: sweeet, a test :)

**Review by anuejn — COMMENTED** [review]
a few nitpicks but looks very nice overall

File: app/src/pages/Editor/ExportDocumentDialog.tsx [hunk 4]
```diff
@@ -1,109 +1,105 @@
...
+];
 
 export function ExportDocumentDialog(): JSX.Element {
   const dispatch = useDispatch();
   const store = useStore();
-  const documentPath = useSelector((state: RootState) => state.editor.present?.path || '');
+  const [home, setHome] = useState('');
+  getHomePath(setHome);
+  const documentPath = useSelector(
+    (state: RootState) => state.editor.present?.path || path.join(home, 'Untitled.audapolis')
```
[the remark this answers is not in the document]
Comment (anuejn) [reply app/src/pages/Editor/ExportDocumentDialog.tsx:66]: ah nvm i got wrong what this does

**Comment by anuejn** [comment]
works nicely :)

File: app/src/core/document.ts [hunk 5]
```diff
@@ -219,6 +219,32 @@ export class DocumentGenerator<
     return new GeneratorBox(renderItemsFromDocumentGenerator(this));
   }
 
+  toTimedParagraphs(): Paragraph<TimedParagraphItem>[] {
+    const timedParagraphs = [];
+    let lastParagraph = null;
+    for (const item of this) {
+      const generatorItemToParagraphItem = (item: DocumentGeneratorItem): TimedParagraphItem => {
+        // eslint-disable-next-line unused-imports/no-unused-vars
+        const { paragraphUuid, itemIdx, speaker, ...rest } = item;
```
[the remark this answers is not in the document]
Comment (pajowu) [reply app/src/core/document.ts:228]: see new issue

File: app/src/core/ffmpeg.ts [hunk 6]
```diff
@@ -5,13 +5,41 @@ import Fessonia from '@tedconf/fessonia';
 import ffmpegPath from 'ffmpeg-static';
 import AppDirectory from 'appdirectory';
 import { player } from './player';
+import { WebVtt } from '../util/WebVtt';
```
[the remark this answers is not in the document]
Comment (pajowu) [reply app/src/core/ffmpeg.ts:8]: as discussed this will be moved to an external library

File: app/src/core/webvtt.ts (hunk 1 above)
Comment (pajowu) [reply app/src/core/webvtt.ts:93]: oops 🧹

**Review by anuejn — APPROVED** [review]
Nice work :)

**Pull request merged**